arch/arm/core/aarch32/prep_c.c

```diff
@@ -144,7 +144,11 @@ static inline void z_arm_floating_point_init(void)
 	__ISB();
 
 	/* Initialize the Floating Point Status and Control Register. */
+#if defined(ARMV8_1_M_MAINLINE)
```

Comment:
```suggestion
#if defined(CONFIG_ARMV8_1_M_MAINLINE)
```

Reply:
Eeek, feel silly that I missed that. Thank @stephanosio.

Reply:
Sorry, miss that too.
BTW, I am thinking something like `__set_FPSCR(FPU_FPDSCR_INIT)` and in some header file, using
```
#if defined(CONFIG_ARMV8_1_M_MAINLINE)
#define FPU_FPDSCR_INIT FPU_FPDSCR_LTPSIZE_Msk
#else
#define FPU_FPDSCR_INIT 0
#endif
```

Reply:
I agree with this format.... but `FPU_FPDSCR_LTPSIZE_Msk` is defined as `(7UL << FPU_FPDSCR_LTPSIZE_Pos)` which would set it to an undefined value. 

```
#if defined(CONFIG_ARMV8_1_M_MAINLINE)
/* With ARMv8.1, LTPSIZE is to be set to "Tail predication not applied" */
#define FPU_FPDSCR_INIT (4 << FPU_FPDSCR_LTPSIZE_Pos)
#else
#define FPU_FPDSCR_INIT 0
#endif
```

Reply:
> but `FPU_FPDSCR_LTPSIZE_Msk` is defined as `(7UL << FPU_FPDSCR_LTPSIZE_Pos)` which would set it to an undefined value.

Indeed, for some reason, I thought this was a single bit field completely ignoring `4`.

Reply:
It's not quite clear to me where the best place for this header to be would be... Ideally, I would expect a reset value to defined in CMSIS, but nothings there for it.